lib/t/calls.ex

```diff
@@ -317,16 +332,45 @@ defmodule T.Calls do
   @spec voicemail_save_message(uuid, uuid, uuid) ::
           {:ok, %Voicemail{}} | {:error, reason :: String.t()}
   def voicemail_save_message(caller_id, match_id, s3_key) do
-    with :ok <- ensure_voicemail_allowed(caller_id, match_id) do
+    with {:ok, mate} <- ensure_voicemail_allowed(caller_id, match_id) do
       voicemail = %Voicemail{caller_id: caller_id, match_id: match_id, s3_key: s3_key}
 
+      push_job =
+        DispatchJob.new(%{
+          "type" => "voicemail_sent",
+          "match_id" => match_id,
+          "caller_id" => caller_id,
+          "receiver_id" => mate
+        })
+
       {:ok, %{voicemail: voicemail}} =
         Multi.new()
         |> Multi.delete_all(:contacts, where(MatchContact, match_id: ^match_id))
         |> Multi.delete_all(:timeslots, where(Timeslot, match_id: ^match_id))
         |> Multi.insert(:voicemail, voicemail)
+        # TODO maybe add by_user_id to match_events table to easier determine voice exchange timer reset?
+        |> Multi.insert(:match_event, %MatchEvent{
+          timestamp: DateTime.truncate(DateTime.utc_now(), :second),
+          match_id: match_id,
+          event: "voicemail_sent"
```

Comment:
or `voicemail_sent_1` `voicemail_sent_2` depending if the current user is 1 or 2

Reply:
relevant https://github.com/getsince/test3/pull/494/files#r777043720